### ISSUE TITLE
broken container cause list containers error.

### ISSUE THREAD

**Issue by coding4m** [docker/swarm#1307]
container withoutname (rm occurs error) cause call 
/containers/json?all=1 occurs error at 

api/handlers.go:197
```javascript
if !filters.Match("name", strings.TrimPrefix(container.Names[0], "/")) {
			continue
		}
```

**Comment by chanwit** [comment]
@coding4m it is unusual that a container does not have name.

Could you explain in which situation your container does not get its name set?

cc @vieux

**Comment by chanwit** [comment]
Sorry for not reading it clearly.

It seems we need fixes to prevent this Names[0] in several places.

**Comment by vieux** [comment]
I don't get it, a container should always have a name

**Comment by cpuguy83** [comment]
This was a bug in engine, it's not possible anymore as of 1.9.
Basically, when a container is partially removed due to an error, the names are the first thing that get wiped out (all of the names)... so we end up with this container with no names.

**Comment by cpuguy83** [comment]
p.s. the partial removal is still possible, but engine won't send a null array anymore.

**Comment by vieux** [comment]
@cpuguy83 I see thanks.

**Comment by abronan** [comment]
Fixed by #1465
